vm/src/primitives.rs

```diff
@@ -246,164 +248,167 @@ pub fn load(vm: &Thread) -> Result<()> {
-        trim => primitive!(1 string_prim::trim),
-        trim_left => primitive!(1 string_prim::trim_left),
-        trim_right => primitive!(1 string_prim::trim_right),
-        compare => named_primitive!(2, "string_prim.compare", string_prim::cmp),
-        append => named_primitive!(2, "string_prim.append", string::append),
-        eq => named_primitive!(2, "string_prim.eq", <str as PartialEq>::eq),
-        slice => named_primitive!(3, "string_prim.slice", string::slice),
-        from_utf8 =>
-            primitive::<fn(Vec<u8>) -> StdResult<String, ()>>("string_prim.from_utf8",
-                                                              string::from_utf8),
-        char_at => named_primitive!(2, "string_prim.char_at", string::char_at),
-        as_bytes => primitive!(1 string_prim::as_bytes)
-    ),
+        record! {
+            len => primitive!(1 string_prim::len),
+            is_empty => primitive!(1 string_prim::is_empty),
+            split_at => primitive!(2 string_prim::split_at),
+            find => named_primitive!(2, "string_prim.find", string_prim::find::<&str>),
+            rfind => named_primitive!(2, "string_prim.rfind", string_prim::rfind::<&str>),
+            starts_with => named_primitive!(2, "string_prim.starts_with", string_prim::starts_with::<&str>),
+            ends_with => named_primitive!(2, "string_prim.ends_with", string_prim::ends_with::<&str>),
+            trim => primitive!(1 string_prim::trim),
+            trim_left => primitive!(1 string_prim::trim_left),
+            trim_right => primitive!(1 string_prim::trim_right),
+            compare => named_primitive!(2, "string_prim.compare", string_prim::cmp),
+            append => named_primitive!(2, "string_prim.append", string::append),
+            eq => named_primitive!(2, "string_prim.eq", <str as PartialEq>::eq),
+            slice => named_primitive!(3, "string_prim.slice", string::slice),
+            from_utf8 => primitive::<fn(Vec<u8>) -> StdResult<String, ()>>("string_prim.from_utf8", string::from_utf8),
+            char_at => named_primitive!(2, "string_prim.char_at", string::char_at),
+            as_bytes => primitive!(1 string_prim::as_bytes)
+        },
     )?;
     vm.define_global(
-        "char",
-        record!(
-        is_digit => primitive!(2 char::is_digit),
-        to_digit => primitive!(2 char::to_digit),
-        len_utf8 => primitive!(1 char::len_utf8),
-        len_utf16 => primitive!(1 char::len_utf16),
-        is_alphabetic => primitive!(1 char::is_alphabetic),
-        is_lowercase => primitive!(1 char::is_lowercase),
-        is_uppercase => primitive!(1 char::is_uppercase),
-        is_whitespace => primitive!(1 char::is_whitespace),
-        is_alphanumeric => primitive!(1 char::is_alphanumeric),
-        is_control => primitive!(1 char::is_control),
-        is_numeric => primitive!(1 char::is_numeric)
-    ),
+        "char_prim",
```

Comment:
Serialization requires that the name of the binding (`char::is_digit` is mapped to `char.is_digit`) matches what can be found in the global environment.

So here it fails because it tries to find `char.is_digit` but it fails because `char_prim` exists but not `char`. Easiest way to fix is to add `use std::char as char_prim` and rename the bindings below here to `char_prim::is_digit` etc. Long term we need a better solution to this `mod_prim + mod` naming.

Reply:
Or rename it back to `char` since that shouldnt cause any collision